bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/PageImpl.java

```diff
@@ -0,0 +1,451 @@
+
+        return wrapper;
+    }
+
+
+    // style helpers zone
+
+    /**
+     * Returns the first numeric selector. The default value is 0
+     *
+     * @return {@link int} the defined traversal depth or 0 if none defined
+     */
+    protected static int getPageTreeTraversalDepth(Style style) {
+        // Depth of the tree of pages
+        final String PN_STRUCTURE_DEPTH = "structureDepth";
+
+        Integer pageTreeTraversalDepth = null;
+
+        if (style != null) {
+            pageTreeTraversalDepth = style.get(PN_STRUCTURE_DEPTH, Integer.class);
+        }
+
+        if (pageTreeTraversalDepth == null) {
+            return 0;
+        }
+
+        return pageTreeTraversalDepth;
+    }
+
+    /**
+     * Get request's entry point attribute value
+     */
+    protected static com.day.cq.wcm.api.Page requestGetHierarchyEntryPoint(@Nonnull SlingHttpServletRequest request) {
+        return (com.day.cq.wcm.api.Page) request.getAttribute(ATTR_HIERARCHY_ENTRY_POINT_PAGE);
+    }
+
+    /**
+     * Returns the page structure patterns to filter the child pages to be exported.
+     * The patterns can either be stored on the template policy of the page or provided as a request parameter
+     *
+     * @param request request
+     * @return {@link List} list of page structure patterns
+     */
+    @Nonnull
+    protected static List<Pattern> getStructurePatterns(@Nonnull SlingHttpServletRequest request, Style currentStyle) {
+        // List of Regexp patterns to filter the exported tree of pages
+        final String PN_STRUCTURE_PATTERNS = "structurePatterns";
+
+        RequestParameter pageFilterParameter = request.getRequestParameter(PN_STRUCTURE_PATTERNS.toLowerCase());
+
+        String rawPageFilters = null;
+
+        if (pageFilterParameter != null) {
+            rawPageFilters = pageFilterParameter.getString();
+        }
+
+        if (currentStyle != null && StringUtils.isBlank(rawPageFilters)) {
+            rawPageFilters = currentStyle.get(PN_STRUCTURE_PATTERNS, String.class);
+        }
+
+        if (StringUtils.isBlank(rawPageFilters)) {
+            return Collections.emptyList();
+        }
+
+        String[] pageFilters = rawPageFilters.split(",");
+
+        List<Pattern> pageFilterPatterns = new ArrayList<>();
+        for (String pageFilter : pageFilters) {
+            pageFilterPatterns.add(Pattern.compile(pageFilter));
+        }
+
+        return pageFilterPatterns;
+    }
+
+    /**
+     * Returns a flat list of all the child pages of a given page
+     *
+     * @param page page from which to extract child pages
+     * @param slingRequest request
+     * @param structurePatterns patterns to filter child pages
+     * @param depth depth of the traversal
+     * @return {@link List}
+     */
+    @Nonnull
+    private List<com.day.cq.wcm.api.Page> getChildPageRecursive(com.day.cq.wcm.api.Page page, SlingHttpServletRequest slingRequest, List<Pattern> structurePatterns, int depth) {
```

Comment:
I'd call it `getTreePages` or `getDescendantPages` unless I misunderstood the intention of this method.